src/System.Threading.Thread/tests/ThreadTests.netstandard.cs

```diff
@@ -20,10 +20,34 @@ public static class ThreadTests
         [Fact]
         public static void ConstructorTest()
         {
-            var t = new Thread(() => { });
-            t = new Thread(() => { }, 0);
-            t = new Thread(state => { });
-            t = new Thread(state => { }, 0);
+            Action<Thread> startThreadAndJoin =
+                t =>
+                {
+                    t.IsBackground = true;
+                    t.Start();
+                    Assert.True(t.Join(UnexpectedTimeoutMilliseconds));
+                };
+            Action<int> verifyStackSize =
+                stackSizeBytes =>
+                {
+                    // Try to stack-allocate an array to verify that close to the expected amount of stack space is actually
+                    // available
+                    int bufferSizeBytes = Math.Max(16 << 10, stackSizeBytes - (64 << 10));
+                    unsafe
+                    {
+                        byte* buffer = stackalloc byte[bufferSizeBytes];
+                        buffer[0] = 0xff;
+                        buffer[bufferSizeBytes - 1] = 0xff;
```

Comment:
Maybe better to use Volatile.Write to make sure that it won't get optimized out.